### PR TITLE
[quant][api] Add assert for backend in get_default_qconfig related apis

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86259

Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

### CODE REVIEW

File: torch/ao/quantization/qconfig.py [hunk 1]
```diff
@@ -228,12 +228,19 @@ def get_default_qconfig(backend='x86', version=0):
...
-      * `backend`: a string representing the target backend. Currently supports
+      * `backend` (str): a string representing the target backend. Currently supports
         `x86` (default), `fbgemm`, `qnnpack` and `onednn`.
 
     Return:
         qconfig
     """
+    supported_backends = ["fbgemm", "x86", "qnnpack", "onednn"]
+    if backend not in supported_backends:
+        raise AssertionError(
```
Comment (andrewor14) [inline torch/ao/quantization/qconfig.py:239]: Should this be `ValueError` instead?

**Review by andrewor14 — APPROVED** [review]

File: torch/ao/quantization/qconfig.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/qconfig.py:239]: I actually checked this as well, ValueError means the type is correct and value is not correct, but in this case the type may not be correct either so I guess AssertionError is OK. we are also using AssertionError for the version number as well

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

**Comment by jerryzh168** [comment]
closed in favor of https://github.com/pytorch/pytorch/pull/87331

**Pull request closed without merging**